File: src/main/java/com/lab/reservation/entity/Device.java
package com.lab.reservation.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName(value = "device", autoResultMap = true)
public class Device extends BaseEntity {
    private String name;
    private Long categoryId;
    private Long labId;
    private String brand;
    private String model;
    private String specs;
    private String imageUrl;
    private String status;
    private Integer needApproval;
    private BigDecimal maxReservationHours;
    private BigDecimal pricePerHour;
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> tags;
    private String description;
}

File: frontend/src/types/device.ts
import type { Page } from './common'

/** 设备状态枚举（与后端 DeviceStatus 一致）。 */
export type DeviceStatus = 'IDLE' | 'IN_USE' | 'MAINTENANCE'

/** 后端 DeviceVO（vo/device/DeviceVO.java）。 */
export interface DeviceVO {
  id: number
  name: string
  categoryId: number | null
  categoryName?: string
  labId: number | null
  labName?: string
  brand?: string
  model?: string
  specs?: string
  imageUrl?: string
  status: DeviceStatus
  /** 0/1 — 是否需审批 */
  needApproval: number
  maxReservationHours?: number | string
  pricePerHour?: number | string
  tags?: string[]
  description?: string
  createdAt?: string
  updatedAt?: string
}

/** 后端 DeviceCalendarItemVO（vo/device/DeviceCalendarItemVO.java）。 */
export interface DeviceCalendarItemVO {
  /** ISO date: yyyy-MM-dd */
  date: string
  slotIndex: number
  reservationId: number
  /** PENDING / APPROVED / IN_USE */
  status: string
}

/** 后端 DeviceCategoryNodeVO（vo/device/DeviceCategoryNodeVO.java）。 */
export interface DeviceCategoryNodeVO {
  id: number
  name: string
  parentId: number
  sort?: number
  children: DeviceCategoryNodeVO[]
}

/** 设备多条件检索参数（对齐 dto/device/DeviceQueryDTO）。 */
export interface DeviceQuery {
  page?: number
  size?: number
  keyword?: string
  categoryId?: number
  labId?: number
  status?: DeviceStatus | ''
  /** 0/1 */
  needApproval?: number
  minPrice?: number | string
  maxPrice?: number | string
}

export type { Page }
